scripts/Muon/GUI/ElementalAnalysis2/auto_widget/ea_auto_tab_view.py

```diff
@@ -0,0 +1,229 @@
+        self.min_width_line_edit = QtWidgets.QLineEdit(self)
+
+        self.max_width_label = QtWidgets.QLabel(" Maximum peak width (KeV) ", self)
+        self.max_width_line_edit = QtWidgets.QLineEdit(self)
+
+        self.estimate_width_label = QtWidgets.QLabel(" Estimate peak width (KeV) ", self)
+        self.estimate_width_line_edit = QtWidgets.QLineEdit(self)
+
+        self.find_peaks_parameter_hlayout3.addWidget(self.min_width_label)
+        self.find_peaks_parameter_hlayout3.addWidget(self.min_width_line_edit)
+        self.find_peaks_parameter_hlayout3.addWidget(self.max_width_label)
+        self.find_peaks_parameter_hlayout3.addWidget(self.max_width_line_edit)
+        self.find_peaks_parameter_hlayout3.addWidget(self.estimate_width_label)
+        self.find_peaks_parameter_hlayout3.addWidget(self.estimate_width_line_edit)
+
+        self.show_peaks_table_combobox = QtWidgets.QComboBox(self)
+        self.show_peaks_table_button = QtWidgets.QPushButton(" Show peaks ", self)
+
+        self.show_peaks_hlayout.addWidget(self.show_peaks_table_combobox)
+        self.show_peaks_hlayout.addWidget(self.show_peaks_table_button)
+
+        self.show_match_table_combobox = QtWidgets.QComboBox(self)
+        self.show_match_table_button = QtWidgets.QPushButton(" Show matches ", self)
+
+        self.show_matches_hlayout.addWidget(self.show_match_table_combobox)
+        self.show_matches_hlayout.addWidget(self.show_match_table_button)
+
+    def setup_buttons(self):
+        self.find_peaks_button.clicked.connect(self.find_peaks_notifier.notify_subscribers)
+        self.show_peaks_table_button.clicked.connect(self.show_peaks_table_notifier.notify_subscribers)
+        self.show_match_table_button.clicked.connect(self.show_match_table_notifier.notify_subscribers)
+        self.clear_match_table_button.clicked.connect(self.clear_match_table_notifier.notify_subscribers)
+        self.default_peak_checkbox.clicked.connect(self.default_peak_checkbox_changed)
+
+    def get_parameters_for_find_peaks(self):
+        parameters = {}
+        try:
+            parameters["min_energy"] = float(self.min_energy_line_edit.text())
+            parameters["max_energy"] = float(self.max_energy_line_edit.text())
+            parameters["threshold"] = float(self.threshold_line_edit.text())
+            parameters["default_width"] = True
+            if not self.default_peak_checkbox.isChecked():
+                parameters["default_width"] = False
+                parameters["min_width"] = float(self.min_width_line_edit.text())
+                parameters["max_width"] = float(self.max_width_line_edit.text())
+                parameters["estimate_width"] = float(self.estimate_width_line_edit.text())
+                if parameters["min_width"] > parameters["max_width"] and \
+                   parameters["min_width"] > parameters["estimate_width"] > parameters["max_width"]:
+                    raise ValueError()
```

Comment:
Could we separate this into two separate if statements and then raise an appropriate error message inside `ValueError`? And also an error message in the if statement below would be useful for a user to understand why a parameter is invalid